docs/semantics/array.md

```diff
@@ -0,0 +1,51 @@
+# Array Semantic
+
+## Overview
+
+An array type is an expression to store a set of values. 
+An array expression type is declared using `array` identifier, is bounded by `[<key_type>,<value_type>]` and is followed by an array name. The two arguments in `[]` are separate by `,`. The data structure `array` type is mutable.
+
+## Syntax
+
+An `array` type declaration in general takes the form of `array[<key_type>, <value_type>] arr;`, where `<key_type>` is the type of key, and `<value_type>` is the type of value.
+

```

Comment:
i know there is an example section at the bottom, but might it be worthwhile to give an example of array declaration here as well?

Reply:
solve